src/eth_rpc/api/eth_api.rs

```diff
@@ -33,11 +33,19 @@ pub trait EthApi {
 
     /// Returns information about a block by hash.
     #[method(name = "getBlockByHash")]
-    async fn block_by_hash(&self, hash: B256, full: bool) -> Result<Option<RichBlock>>;
+    async fn block_by_hash(
```

Comment:
can't we use the same types as reth?
https://github.com/paradigmxyz/reth/blob/v1.0.7/crates/rpc/rpc/src/engine.rs#L105

Reply:
Theoretically we could but it's a generic that has to implement something related to the network and so we don't really need it.

I tried to implement it but it was quite heavy and not necessarily useful for us because we always use the same thing so we don't necessarily need generic.

But I'm going to open a follow up issue right now to investigate these heavy types.

Reply:
Opened an issue here https://github.com/kkrt-labs/kakarot-rpc/issues/1398